pym/bob/scm/git.py

```diff
@@ -81,25 +91,61 @@ def getProperties(self):
                     ("refs/heads/" + self.__branch))
             )
         }]
+        for key, val in self.__remotes.items():
+            properties[0].update({GitScm.REMOTE_PREFIX+key : val})
+        return properties
 
     def asScript(self):
+        remotes_array = """
+# create an array of all remotes for this repository
+declare -A BOB_GIT_REMOTES=( [origin]="{URL}" )""".format(URL=self.__url)
+        if self.__remotes:
+            # add additional remotes to array
+            for name, url in self.__remotes.items():
+                remotes_array += """
+BOB_GIT_REMOTES[{NAME}]="{URL}" """.format(NAME=name, URL=url)
+        # create script to handle remotes
+        remotes_check_add_all = remotes_array + """
+# remove remotes from array that are already known to Git
+while read -r REMOTE_NAME ; do
+    # check for empty variable in case no remote at all is specified
+    if [ -z "$REMOTE_NAME" ]; then
+        continue
+    fi
+    # check if URL has changed
+    if [ ! "${BOB_GIT_REMOTES[$REMOTE_NAME]}" == "$(git remote get-url $REMOTE_NAME)" ]; then
+        git remote set-url "$REMOTE_NAME" "${BOB_GIT_REMOTES[$REMOTE_NAME]}"
+    fi
+    # check if existing remote is configured
+    if [ "${BOB_GIT_REMOTES[$REMOTE_NAME]+_}" ]; then
+        # it is configured, therefore no need to keep in list
+        unset "BOB_GIT_REMOTES[$REMOTE_NAME]"
+    else
+        # not configured but existing, therefore remove it
+        git remote remove "$REMOTE_NAME"
```

Comment:
I don't think this is a good idea. Previously it was perfectly ok for the user to add any remotes he likes and Bob would not touch it. Now we would remove them. As this might break some users work flows I would rather not touch them...

Reply:
Or you have to distinguish somehow between an unknown remote and one that has been configured before but is now removed from the configuration. The latter could be safely deleted.

Reply:
You're right. Maybe the `.git` directory could be abused to store the list of remotes that were configured by Bob. But I guess it will add quite some logic...

Reply:
Sounds complicated. I'd rather not implement that. Removed removing of remotes. :)

Reply:
:smile: Ok for me. If there's a need it can be implemented later...